sql/migrate/lex_test.go

```diff
@@ -105,3 +105,57 @@ cmd7;
 	require.Equal(t, "cmd7;", stmts[7].Text)
 	require.Equal(t, []string{""}, stmts[7].Directive("nolint"))
 }
+
+func TestLex_Errors(t *testing.T) {
+	for _, tt := range []struct {
+		name, stmt, err string
+	}{
+		{
+			name: "unclosed single at 1:1",
+			stmt: "'this quote is unclosed at 1:1",
+			err:  "1:1: unclosed quote '\\''",
+		},
+		{
+			name: "unclosed single at 1:6",
+			stmt: "12345'this quote is unclosed at pos 7",
+			err:  "1:6: unclosed quote '\\''",
+		},
+		{
+			name: "unclosed single at EOS",
+			stmt: "unclosed '",
+			err:  "1:10: unclosed quote '\\''",
+		},
+		{
+			name: "unclosed double at 1:1",
+			stmt: `"unclosed double`,
+			err:  `1:1: unclosed quote '"'`,
+		},
+		{
+			name: "unclosed double at 2:2",
+			stmt: `unclosed double at 2:2\n "`,
+			err:  `2:2: unclosed quote '"'`,
+		},
+		{
+			name: "unclosed double at 5:5",
+			stmt: `unclosed double at 2:2\n\n\n\n    "`,
```

Comment:
```suggestion
			stmt: "unclosed double at 2:2\n\n\n\n    "",
```
\n inside raw strings are escaped